ui/src/app/pages/admin/workspace/cloud-environments-table.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+import { useState } from 'react';
+import { Column } from 'primereact/column';
+import { DataTable } from 'primereact/datatable';
+
+import {
+  AdminWorkspaceResources,
+  ListRuntimeResponse,
+  UserAppEnvironment,
+} from 'generated/fetch';
+
+import {
+  fromRuntimeStatus,
+  fromUserAppStatusWithFallback,
+  toUIAppType,
+  UIAppType,
+} from 'app/components/apps-panel/utils';
+import { Button } from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import { TooltipTrigger } from 'app/components/popups';
+import { workspaceAdminApi } from 'app/services/swagger-fetch-clients';
+import { getCreator } from 'app/utils/runtime-utils';
+
+interface CloudEnvironmentRow {
+  appType: UIAppType;
+  name: string;
+  creator: string;
+  createdTime: string;
+  lastAccessedTime: string;
+  status: string;
+}
+
+const runtimeToRow = (runtime: ListRuntimeResponse): CloudEnvironmentRow => {
+  const { runtimeName, createdDate, dateAccessed, status } = runtime;
+  return {
+    appType: UIAppType.JUPYTER,
+    name: runtimeName,
+    creator: getCreator(runtime),
+    createdTime: new Date(createdDate).toDateString(),
+    lastAccessedTime: new Date(dateAccessed).toDateString(),
+    status: fromRuntimeStatus(status),
+  };
+};
+
+const userAppToRow = (userApp: UserAppEnvironment): CloudEnvironmentRow => {
+  const { appType, appName, creator, createdDate, dateAccessed, status } =
+    userApp;
+  return {
+    appType: toUIAppType[appType],
+    name: appName,
+    creator,
+    createdTime: new Date(createdDate).toDateString(),
+    lastAccessedTime: new Date(dateAccessed).toDateString(),
+    status: fromUserAppStatusWithFallback(status),
+  };
+};
+
+interface Props {
+  resources: AdminWorkspaceResources;
+  workspaceNamespace: string;
+  onDelete: () => void;
+}
+export const CloudEnvironmentsTable = ({
+  resources: { runtimes, userApps },
+  workspaceNamespace,
+  onDelete,
+}: Props) => {
+  const [runtimeToDelete, setRuntimeToDelete] = useState<string>();
+  const [confirmDeleteRuntime, setConfirmDeleteRuntime] = useState(false);
+
+  const deleteRuntime = () => {
+    setConfirmDeleteRuntime(false);
+    workspaceAdminApi()
+      .deleteRuntimesInWorkspace(workspaceNamespace, {
+        runtimesToDelete: [runtimeToDelete],
+      })
+      .then(() => {
+        setRuntimeToDelete(null);
+        onDelete();
+      });
+  };
+
+  const cancelDeleteRuntime = () => {
+    setConfirmDeleteRuntime(false);
+    setRuntimeToDelete(null);
+  };
+
+  const DeleteButton = ({ row }: { row: CloudEnvironmentRow }) => (
+    <TooltipTrigger
+      content='Deletion is currently only availble for Jupyter.  See RW-8943.'
```

Comment:
Normally I don't know if we'd want to expose JIRA ticket IDs to users ... but this is an admin page, so that feels OK

Reply:
Agreed, I think this is fine.